installer/conf/health_model_definition.json

```diff
@@ -0,0 +1,248 @@
+        "parent_monitor_id": "all_workloads",
+        "labels": [
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ]
+    },
+    {
+        "monitor_id": "all_workloads",
+        "parent_monitor_id": "cluster",
+        "labels": [
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ]
+    },
+    {
+        "monitor_id": "node_cpu_utilization",
+        "parent_monitor_id": "node",
+        "labels": [
+            "kubernetes.io/hostname",
+            "agentpool",
+            "kubernetes.io/role",
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ]
+    },
+    {
+        "monitor_id": "node_memory_utilization",
+        "parent_monitor_id": "node",
+        "labels": [
+            "kubernetes.io/hostname",
+            "agentpool",
+            "kubernetes.io/role",
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ]
+    },
+    {
+        "monitor_id": "node_condition",
+        "parent_monitor_id": "node",
+        "labels": [
+            "kubernetes.io/hostname",
+            "agentpool",
+            "kubernetes.io/role",
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ]
+    },
+    {
+        "monitor_id": "node",
+        "aggregation_algorithm": "worstOf",
+        "labels": [
+            "kubernetes.io/hostname",
+            "agentpool",
+            "kubernetes.io/role",
+            "container.azm.ms/cluster-region",
+            "container.azm.ms/cluster-subscription-id",
+            "container.azm.ms/cluster-resource-group",
+            "container.azm.ms/cluster-name"
+        ],
+        "parent_monitor_id": [
+            {
+                "label": "kubernetes.io/role",
+                "operator": "==",
+                "value": "master",
+                "id": "master_node_pool"
+            },
+            {
+                "label": "kubernetes.io/role",
+                "operator": "==",
+                "value": "agent",
+                "id": "agent_node_pool"
+            }
+        ]
+    },
+    {
+        "monitor_id": "master_node_pool",
+        "aggregation_algorithm": "percentage",
+        "aggregation_algorithm_params": {
+            "critical_threshold": 80.0,
```

Comment:
does this mean this threshold is applicable for number of nodes in the pool ? i.e 80% of nodes for any monitor in the pool is above threshold, for the pool to roll up to 'read' state? I see threshold in 2 places (definition & config). Is that intended ?

Reply:
If we are rolling up to nodepool, did u validate in multi pool scenarios (which is under preview in AKS and will goto GA in a month or so..)?

Reply:
we are rolling up to nodepool, and there shouldnt be any difference w.r.t the rollup between one pool or multiple pools.

Reply:
80% means, if less than 80% of the nodes are not in green, this aggregate monitor will be in 'Critical' State

Reply:
configurations are monitor level configurations. 
Definitions are an implementation detail for how the monitor is aggregated. in the future, both will be opened up for configuration by customer